planner/core/expression_rewriter.go

```diff
@@ -1375,7 +1375,8 @@ func (er *expressionRewriter) inToExpression(lLen int, not bool, tp *types.Field
 		er.ctxStackAppend(expression.NewNull(), types.EmptyName)
 		return
 	}
-	if leftEt == types.ETInt {
+	containMut := expression.ContainMutableConst(er.sctx, args)
+	if !containMut && leftEt == types.ETInt {
```

Comment:
Can you please add a test case to cover this bug fix? i.e, manufacture a case to reproduce wrong query result caused by this `RefineComparedConstant` when plan cache is enabled.

Reply:
drop table if exists t1;
    create table t1(c1 tinyint unsigned);
    insert into t1 values(111);
    prepare stmt from 'select 1 from t1 where c1 in (?)';
    set @a = '1.1';
    execute stmt using @a;
    set @a = '111';
    execute stmt using @a;

Above case can cover this situation. Added in TestConstPropAndPPDWithCache()